File: src/Gastromio.Persistence.MongoDB/CustomerInfoModel.cs
namespace Gastromio.Persistence.MongoDB
{
    public class CustomerInfoModel
    {
        public string GivenName { get; set; }
        public string LastName { get; set; }
        public string Street { get; set; }
        public string AddAddressInfo { get; set; }
        public string ZipCode { get; set; }
        public string City { get; set; }
        public string Phone { get; set; }
        public string Email { get; set; }
    }
}

File: src/Gastromio.App/ClientApp/src/app/order/models/customer-info.model.ts
export class CustomerInfoModel {

  constructor(init?: Partial<CustomerInfoModel>) {
    if (init) {
      Object.assign(this, init);
    }
  }

  public givenName: string;

  public lastName: string;

  public street: string;

  public addAddressInfo: string;

  public zipCode: string;

  public city: string;

  public phone: string;

  public email: string;

  public clone(): CustomerInfoModel {
    return new CustomerInfoModel({
      givenName: this.givenName,
      lastName: this.lastName,
      street: this.street,
      addAddressInfo: this.addAddressInfo,
      zipCode: this.zipCode,
      city: this.city,
      phone: this.phone,
      email: this.email
    });
  }

}
